dbt_meshify/storage/file_manager.py

```diff
@@ -32,74 +31,72 @@ def dump(self, data, stream=None, **kw):
 
 yaml = DbtYAML()
 yaml.register_class(Identifier)
-FileContent = Union[Dict[str, str], str]
 
 
-class BaseFileManager(ABC):
-    @abc.abstractmethod
-    def read_file(self, path: Path) -> Union[Dict[str, Any], str, None]:
+class FileManager(Protocol):
```

Comment:
total curiosity question: what's a `Protocol`? similar to an abstract base class?

Reply:
A Protocol is similar to an Abstract Base Class, but it uses duck typing instead of inheritance. So, for instance, we don't need to have a bunch of file managers inheriting from an abstract file manager. Instead, we can define a protocol, and _any_ class that implements the correct methods will be allowed via the type checker. It's rather similar to the trait system in rust, which I LOVE.